sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -761,7 +775,22 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
             triggerIngestion(groupState);
         } else if (pendingLogCount > 0 && !groupState.mScheduled) {
             groupState.mScheduled = true;
-            mAppCenterHandler.postDelayed(groupState.mRunnable, groupState.mBatchTimeInterval);
+            long delay = groupState.mBatchTimeInterval;
+            if (groupState.mBatchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL) {
+                long now = System.currentTimeMillis();
+                long startTimer = SharedPreferencesManager.getLong(START_TIMER_PREFIX + groupState.mName);
+                if (startTimer == 0) {
+                    SharedPreferencesManager.putLong(START_TIMER_PREFIX + groupState.mName, now);
```

Comment:
no we must always save the start time in case application is killed.

Reply:
@guperrot It's "always". if there is no saved value -> save

Reply:
It is not, you are saving only if there is no save value and that's the only time it happens.
What I was saying is we have to save at every timer invocation.

Reply:
Ok it works because there is a call to remove later, we need more comments to explain how this works.